crates/cli/src/cli_args.rs

```diff
@@ -0,0 +1,95 @@
+pub mod add;
+pub mod install;
+pub mod run;
+pub mod store;
+
+use crate::State;
+use add::AddArgs;
+use clap::{Parser, Subcommand};
+use install::InstallArgs;
+use miette::Context;
+use pacquet_executor::execute_shell;
+use pacquet_npmrc::Npmrc;
+use pacquet_package_json::PackageJson;
+use run::RunArgs;
+use std::{env, path::PathBuf};
+use store::StoreCommand;
+
+/// Experimental package manager for node.js written in rust.
+#[derive(Debug, Parser)]
+#[clap(name = "pacquet")]
+#[clap(bin_name = "pacquet")]
+#[clap(version = "0.2.1")]
+#[clap(about = "Experimental package manager for node.js")]
+pub struct CliArgs {
+    #[clap(subcommand)]
+    pub command: CliCommand,
+
+    /// Set working directory.
+    #[clap(short = 'C', long, default_value = ".")]
+    pub dir: PathBuf,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum CliCommand {
+    /// Initialize a package.json
+    Init,
+    /// Add a package
+    Add(AddArgs),
+    /// Install packages
+    Install(InstallArgs),
+    /// Runs a package's "test" script, if one was provided.
+    Test,
+    /// Runs a defined package script.
+    Run(RunArgs),
+    /// Runs an arbitrary command specified in the package's start property of its scripts object.
+    Start,
+    /// Managing the package store.
+    #[clap(subcommand)]
+    Store(StoreCommand),
+}
+
+impl CliArgs {
+    /// Execute the command
+    pub async fn run(self) -> miette::Result<()> {
+        let CliArgs { command, dir } = self;
+        let package_json_path = || dir.join("package.json");
+        let npmrc = || Npmrc::current(env::current_dir, home::home_dir, Default::default).leak();
+        let state = || State::init(package_json_path(), npmrc()).wrap_err("initialize the state");
```

Comment:
`CliArgs` is already destroyed (i.e. destructured and moved) by the time these functions are called. So it's impossible.

`CliArgs` must be moved because I don't want to clone `self.command`.

And although it is technically possible to also capture every `self` by reference in every cli `run` functions for now, it would restrict future code to only capture by reference.